### PR TITLE
fix required flags and added validate function

### LINKED ISSUE #32: Required flags should be present on request variables

### PR DESCRIPTION
**Pull request by jgrayson676**
fix #32

### CODE REVIEW

File: src/outbound.coffee [hunk 1]
```diff
@@ -356,7 +356,25 @@ parseCollateralIds = (value) ->
...
+# Validate ---------------------------------------------------------------
+#
+
+validate = (vars) ->
+  return 'username must not be blank' unless vars.lendio.username
+  return 'password must not be blank' unless vars.lendio.password
+  return 'first_name must not be blank' unless vars.lead.first_name
+  return 'last_name must not be blank' unless vars.lead.last_name
+  return 'email must be a valid address' unless vars.lead.email and vars.lead.email.valid
+  return 'phone_1 must be a valid phone number' unless vars.lead.phone_1 and vars.lead.phone_1.valid
```
Comment (cgrayson) [inline src/outbound.coffee]: Think this could be very slightly shorter with, `… unless vars.lead.phone_1?.valid`.

File: src/outbound.coffee [hunk 2]
```diff
@@ -356,7 +356,25 @@ parseCollateralIds = (value) ->
...
+  return 'finance amount must be a valid numeric amount' unless parseFinanceAmountId(vars.lead.loan.amount)
+  return 'time in business must be a valid range' unless parseTimeInBusinessId(vars.lead.company.time_in_business)
+  return 'monthly sales must be a valid range' unless parseAverageMonthlySalesId(vars.lead.company.monthly_sales)
+  return 'monthly charges must be a valid range' unless parseMonthlyChargesId(vars.lead.company.monthly_charges)
+  return 'credit score must be a valid three digit score' unless parseCreditScoreId(vars.lead.credit_score)
+  return 'industry must be a valid industry type' unless parseIndustryId(vars.lead.company.industry)
```
Comment (cgrayson) [inline src/outbound.coffee]: This looks good, as long as we're confident that all of these are truly required by Lendio.

Comment (jgrayson676) [inline src/outbound.coffee]: It's correct as far as I know, given the information in google drive. I didn't see anything that seemed obviously inconsistent or wrong, so I have no reason to think it's incorrect.

Comment (cgrayson) [inline src/outbound.coffee]: I see. Also saw [the Lendio API docs](http://www.lendio.com/api/docs/borrowers#create) list all those as required.

**Comment by cgrayson** [comment]
LGTM

**Pull request merged**